module/documents/actor/actor.mjs

```diff
@@ -47,6 +47,17 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
 
   /* -------------------------------------------- */
 
+  /**
+   * Get all origins which have spellcasting ability.
+   * @type {Array<Item5e>}
+   */
+  get spellCastingClasses() {
+    return Object.values(this.classes).filter(({system}) => system.spellcasting
+    && system.spellcasting.progression !== "none" );
+  }
```

Comment:
Rather than returning an array of items, it would probably be more useful to match the structure of the `classes` method and return a similar object:

```suggestion
  /**
   * Get all classes which have spellcasting ability.
   * @type {Record<string, Item5e>}
   */
  get spellcastingClasses() {
    return Object.entries(this.classes).reduce((obj, [identifier, cls]) => {
      if ( cls.spellcasting && (cls.spellcasting.progression !== "none") ) obj[identifier] = cls;
      return obj;
    }, {});
  }
```

This structure would allow the value to be read directly in the template rather than having to undergo additional processing.

We should also make sure we refer to `item.spellcasting` rather than `item.system.spellcasting`, because that allows us to take subclass spellcasting into account also.